File: ruby/grade-school/grade_school.rb
class School
  def initialize
    @school = {}
  end

  def add(name, grade)
    @school[grade] = [] if @school[grade].nil? && grade.to_i > 0
    @school[grade] << name
  end

  attr_reader :school

  def students(grade)
    (school[grade] || []).sort
  end

  def students_by_grade
    school.keys.sort.map{ |k| {grade: k, students: students(k)}}
  end
end

module BookKeeping
  VERSION = 3
end

File: javascript/grade-school/grade-school.js
var School = function() {
  var roster = {};

  this.add = function(student, grade) {
    if (roster[grade] == undefined) roster[grade] = []
    roster[grade].push(student);
  };

  this.grade = function(grade) {
    return (roster[grade] || []).sort();
  };

  this.roster = function() {
    return Object.keys(roster).reduce(function(out,curr) {
      out[curr] = roster[curr].sort();
      return out;
    }, {});
  };
};

module.exports = School;
